### ISSUE TITLE
Complex Portal xrefs that got missed in AC update

### ISSUE THREAD

**Issue by bmeldal** [geneontology/go-ontology#16513]
@tonysawfordebi 
 
Please see xrefs in 
https://www.ebi.ac.uk/QuickGO/term/GO:0090733

They contain the old EBI-xxxxxxx numbers and are mapped to IntAct which gives 0 results - because they are complexes. If you copy/paste the ACs into the CP they are valid.
I think we missed something when we did all the updates.
We should also populate the description with the shortlabel.

In fact, the CP xrefs are generally still missing, e.g.
https://www.ebi.ac.uk/QuickGO/term/GO:0070187
should have a link to https://www.ebi.ac.uk/complexportal/complex/CPX-152

I know there was a bug that stopped the xrefs from showing in the recent version of QuickGO but I thought it was fixed. Evidentially not.

Apologies, if this is not the right tracker.

**Comment by tonysawfordebi** [comment]
We need a GO editor to fix the xrefs in the ontology itself; we were only able to update the identifiers in annotations in our database.

Not sure who's the most appropriate person to tag here, so I'll just pick @pgaudet at random...

**Comment by bmeldal** [comment]
Thanks, Tony. Not sure I believe that tagging Pascale was a random action ;-)

Pascale, or whoever if going to look at this, it may require a skype call and screenshare to explain what's happening - or not happening in this case.

**Comment by ukemi** [comment]
Looks like we have two xref formats:
IntAct:EBI-##########
ComplexPortal:CPX-###

**Comment by bmeldal** [comment]
The latter is the correct format (ComplexPortal:CPX-###) since we changed out ACs. We changed all annotation files and I assumed (how silly of me!) that the xrefs would be captured as well. However, I don't know where QuickGO takes them from, as we used to import them from one of our files, not just the dbxrefs we supply for new terms. There was a pipeline for the old QuickGO site and when the new site came online it didn't work anymore. But there is a placeholder for Complexes in the navigation bar! Maybe we have to produce a new pipeline using one of our new files. ComplexTab would be the easiest.

**Comment by cmungall** [comment]
can the prefix be substituted in the go editors file or is a mapping table required?

**Comment by bmeldal** [comment]
No, the data for the Protein Complex section in QuickGO should come from one of our release files but during the update of QuickGO and the CP release files something broke. I never knew where the old xrefs came from as they got put in before my time. I just know that they disappeared at some point.

I also don't know what happens to the dbxrefs that we add when we request new terms but they are not exhaustive as new complexes get released that map to the same GO terms.

The new GPAD is insufficient as that only contains complexes with full annotations and a lot of historic entries lack the Reference and Evidence fields so are excluded from the GPAD. Here we just need the mapping of CP complex is_a GO complex.

**Comment by tonysawfordebi** [comment]
I've just had a look at the code to remind myself how all this stuff works, and I can now confirm that in QuickGO's term page, the content of the Cross-references section comes from what we find in the ontology itself, while the content of the Protein Complexes section is derived from annotations between ComplexPortal IDs and GO terms that are descendants of GO:0032991. In this latter category we have a grand total of 26 entries, namely:

GO ID|GO Term|ComplexPortal ID|Complex Name
-----|-------|----------|------------
GO:0000333|telomerase catalytic core complex|CPX-17|Telomerase catalytic core complex
GO:0000333|telomerase catalytic core complex|CPX-19|Telomerase catalytic core complex
GO:0005669|transcription factor TFIID complex|CPX-915|General transcription factor complex TFIID
GO:0005697|telomerase holoenzyme complex|CPX-17|Telomerase catalytic core complex
GO:0005890|sodium:potassium-exchanging ATPase complex|CPX-57|Sodium:potassium-exchanging ATPase complex
GO:0008076|voltage-gated potassium channel complex|CPX-3073|Voltage-gated potassium channel complex variant 2
GO:0034098|VCP-NPL4-UFD1 AAA ATPase complex|CPX-136|Vcp-Npl4-Ufd1 AAA ATPase complex
GO:0034098|VCP-NPL4-UFD1 AAA ATPase complex|CPX-138|Vcp-Npl4-Ufd1 AAA ATPase complex
GO:0034245|mitochondrial DNA-directed RNA polymerase complex|CPX-3146|Mitochondrial DNA-directed RNA polymerase complex
GO:0034688|integrin alphaM-beta2 complex|CPX-1826|Integrin alphaM-beta2 complex
GO:0036488|CHOP-C/EBP complex|CPX-60|CHOP-C/EBPalpha complex
GO:0070522|ERCC4-ERCC1 complex|CPX-478|ERCC1-XPF endonuclease complex
GO:0070522|ERCC4-ERCC1 complex|CPX-491|Ercc1-Xpf endonuclease complex
GO:0070578|RISC-loading complex|CPX-134|RISC-loading complex, TARBP2 variant
GO:0070877|microprocessor complex|CPX-3080|Microprocessor complex
GO:0071818|BAT3 complex|CPX-132|BAT3 complex
GO:0097123|cyclin A1-CDK2 complex|CPX-2005|Cyclin A1-CDK2 complex
GO:0097124|cyclin A2-CDK2 complex|CPX-2006|Cyclin A2-CDK2 complex
GO:0097134|cyclin E1-CDK2 complex|CPX-2015|Cyclin E1-CDK2 complex
GO:0097135|cyclin E2-CDK2 complex|CPX-2016|Cyclin E2-CDK2 complex
GO:1990590|ATF1-ATF4 transcription factor complex|CPX-7|ATF1-ATF4 transcription factor complex
GO:1990590|ATF1-ATF4 transcription factor complex|CPX-9|ATF1-ATF4 transcription factor complex
GO:1990666|PCSK9-LDLR complex|CPX-128|LDLR-PCSK9 complex
GO:1990667|PCSK9-AnxA2 complex|CPX-130|ANXA2-PCSK9 complex
GO:1990730|VCP-NSFL1C complex|CPX-263|Nsfl1c-Vcp complex
GO:1990879|CST complex|CPX-2129|CST complex

**Comment by bmeldal** [comment]
Ok, far too few. We need to sit down and check what comes from where. I'll email you.

**Comment by pgaudet** [comment]
@bmeldal I suppose we wait for the full list before doing anything here ?

**Comment by pgaudet** [comment]
Question for editors WRT Point 1 above: looks like annotations were used to populate xreferences in the ontology - I was not aware that there was such a process - is this OK? What else get added ?

**Comment by pgaudet** [comment]
Some background: 
- Some annotations using CP entity as an annotation subject and a GO CC-complex as a GO term exist in P2GO (see list of 26 above). These cannot be made anymore. What should we do with existing ones? 
- In CP cross references are added, including GO CC-complex terms. These not integrated in the GPAD file, so are not part of the annotation set that we import. This results in 1515 legit xrefs that are not imported into GO (and that number will keep on increasing as CP annotated more complexes).

**Comment by pgaudet** [comment]
* Proposition for managing dbxreferences have been moved to #16553
* Annotation removal will be moved to the annotation tracker (ie the 26 CP AC to GO term xrefs in the ontology that were annotated in P2GO before 20/4/18)

**Comment by pgaudet** [comment]
@bmeldal When everything has been moved to new tickets, please close. 

Thanks !
